### PR TITLE
Added support for multi-target assign

### PR DESCRIPTION
**Pull request by WorldofKerry**
- `a, b = 1, 2`, etc.
- Improved debugging tools
- [x] Clean up code
- [x] Make recursive for multi-layer tuple unpacking

### CODE REVIEW

File: python2verilog/ir/context.py [hunk 1]
```diff
@@ -271,7 +279,7 @@ def add_global_var(self, var: Var):
         var = typed_strict(var, Var)
         if (
             var in self._global_vars
-            or var in self._input_vars
+            or var in self.input_vars
             or var in self.output_vars
```
Comment (Haxrox) [inline python2verilog/ir/context.py:283]: I would double check this logic to make sure you're accessing the correct variables

File: python2verilog/ir/context.py [hunk 2]
```diff
@@ -298,9 +306,9 @@ def get_strs(variables: Sequence[Var]):
 
         assert isinstance(name, str)
         variables = [
-            *list(get_strs(self._global_vars)),
-            *list(get_strs(self._input_vars)),
-            *list(get_strs(self._output_vars)),
+            *list(get_strs(self.global_vars)),
+            *list(get_strs(self.input_vars)),
+            *list(get_strs(self.output_vars)),
```
Comment (Haxrox) [inline python2verilog/ir/context.py:311]: I would double check this logic to make sure you're accessing the array of variables instead of the function

**Comment by Haxrox** [comment]
> Not sure if I should make `CodegenConfig` apart of `Context` class. Currently only the default `CodegenConfig` should be used by end users.

I'm assuming `CodegenConfig` is something you pass into the `Context` class to configure optimizations, variable names, etc.

If `Context` is not being used by end users, I would keep them separated.

File: tests/utils/test_peek_counter.py
[the remark this answers is not in the document]
Comment (WorldofKerry) [reply tests/utils/test_peek_counter.py:1]: I've removed it.

**Review by Haxrox — APPROVED** [review]

File: python2verilog/frontend/generator.py [hunk 3]
```diff
@@ -404,6 +397,67 @@ def __parse_while(self, stmt: pyast.While, nextt: ir.Element, prefix: str):
...
+            )
+            prev = node.child
+        if last_edge:
+            assert guard(head, ir.AssignNode)
+            assert guard(prev, ir.NonClockedEdge)
+            return head, prev
+        assert guard(head, ir.AssignNode)
+        assert guard(node, ir.AssignNode)
+        return head, node
+
```
Comment (WorldofKerry) [inline python2verilog/frontend/generator.py:460]: Most of the new code

File: python2verilog/frontend/generator.py [hunk 4]
```diff
@@ -96,62 +83,68 @@ def _name_to_var(name: pyast.expr) -> ir.Var:
...
+            for t, v in zip(target.elts, value.elts):
+                yield from self._target_value_visitor(t, v)
+        elif isinstance(target, pyast.Name):
+            if not self._context.is_declared(target.id):
+                self._context.add_global_var(ir.Var(py_name=target.id))
+            yield (ir.Var(target.id), self.__parse_expression(value))
         else:
-            raise TypeError(f"Unsupported lvalue type {type(node)} {pyast.dump(node)}")
-        return self.__parse_expression(node)
+            raise TypeError(f"{pyast.dump(target)} {pyast.dump(value)}")
```
Comment (WorldofKerry) [inline python2verilog/frontend/generator.py:124]: This is new code too

**Pull request merged**